[IMP] accounting/reporting: current year earnings in v19 (#16598)
odoo/documentation: content/applications/finance/accounting/reporting/year_end.rst

.. _accounting/year-end/current-year-earnings:

Current year earnings
~~~~~~~~~~~~~~~~~~~~~

.. important::
   In versions before Odoo 19, current year earnings were handled differently. For consistency in
   your reporting history, upgrading users should revisit past fiscal years and explicitly book the
   results that were previously handled implicitly by the system. These :ref:`manual journal entries
   <accounting/year-end/manual-je>` should be made in chronological order. For more information on
   why this change was made and how versions before Odoo 19 compare to versions after it, see this
   `blog post <https://www.odoo.com/blog/business-hacks-1/how-to-manage-current-years-earnings-in-odoo-19-compared-to-previous-versions-2107>`_.

At the end of a fiscal year, a company will almost always have a net profit or a net loss that is
calculated by summing the income and expense accounts of the profit and loss statement. Odoo
displays the result of the year **N** as a **dynamic line** called "Undistributed Profits/Losses"

> within the trial balance and general ledger of the year **N+1**. To clear this dynamic line, create

Reviewer: ```suggestion
displays the result for year **N** as a **dynamic line** called *Undistributed Profits/Losses*
in the trial balance and general ledger of year **N+1**. To clear this dynamic line, create
```
Not mandatory, just a suggestion to slightly improve clarity.

a :ref:`manual journal entry <accounting/year-end/manual-je>` in year **N** for the end-of-year
result using a :ref:`Current Year Earnings type account <accounting/year-end/appropriation>`.

.. _accounting/year-end/trial-balance:

Balancing the trial balance
***************************

A trial balance is, by default, always balanced (meaning the sum of debits equals the sum of

> credits) because it records the sum of all journal entries which are themselves balanced. However,
> **balance sheet accounts** (assets, liability, and equity) carry over their balances from previous
> fiscal years, whereas **profit and loss accounts** (income, expense, and
> :ref:`appropriation accounts <accounting/year-end/appropriation>`) reset to zero at the start of a

Reviewer: ```suggestion
credits) because it records the sum of all journal entries, which are themselves balanced. However,
**balance sheet accounts** (assets, liabilities, and equity) carry over their balances from previous
fiscal years, whereas **profit and loss accounts** (income, expenses, and
:ref:`appropriation accounts <accounting/year-end/appropriation>`) reset to zero at the start of a
```

new fiscal year. This difference in account behavior would leave the trial balance unbalanced if it

> were not for the dynamic line - "Undistributed Profits/Losses" - that balances this difference

Reviewer: ```suggestion
were not for the dynamic line *Undistributed Profits/Losses* that balances this difference
```
You could use italics instead (looks clearer, but it might be just me)

temporarily until an explicit, :ref:`manual journal entry <accounting/year-end/manual-je>` is made
to appropriate the profit or the loss. Posting an explicit entry to a :ref:`current year earnings
account <accounting/year-end/appropriation>` at year-end "neutralizes" the dynamic line and ensures
the trial balance is balanced by actual journal items. This manual journal entry does not need to be
made immediately following the end of the fiscal year because the trial balance is always balanced,
but whenever the company's profit/loss is ready to be appropriated, this manual journal entry
replaces the dynamic line.

.. _accounting/year-end/appropriation:

Appropriation (current year earnings) accounts
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Appropriation accounts (also called by their account type, current year earnings) are used to
allocate a company's profit or loss, whether that allocation is to reserves, dividends, or any other
destination. There are several key pieces of information to know about these accounts and how they
function:

> - **Annual reset**: Like the income and expense accounts that are also found on the profit

Reviewer: ```suggestion
- **Annual reset**: Like the income and expenses accounts that are also found on the profit
```
Don't we use the plural to refer to the expense**s** account? (i.e., in the product)

  and loss statement, allocation accounts have their opening balances reset at the beginning of a
  fiscal year.
- **P&L presentation**: These accounts appear at the bottom of the **profit and loss** statement.
- **No profit effect**: Unlike income and expense accounts, these accounts **do not affect** the
  calculation of the net profit.
- **Balance sheet counterparts**: Only the *counterparts* of the journal items that debit or credit
  appropriations accounts (e.g., the dividends payable or the reserve accounts) appear on the
  balance sheet. This maintains a strict separation between the process of distributing profit and
  the resulting liabilities or equity.

.. _accounting/year-end/manual-je:

Manual journal entry
^^^^^^^^^^^^^^^^^^^^

> To roll up income and expenses and replace the dynamic "Undistributed Profits/Losses" line with an

Reviewer: ```suggestion
To roll up income and expenses and replace the dynamic *Undistributed Profits/Losses* line with an
```

   way, to fully distribute the year's profit or loss, the amount should sum to the amount that is
   on the dynamic :guilabel:`Undistributed Profits/Losses` line of the trial balance **for the year
   after the profit or loss was incurred**.
#. Add one or more additional lines that credit (in the case of a profit) or debit (in the case of a
   loss) one or more destination accounts to balance the journal entry. Depending on the
   localization and the situation, the destination account(s) will vary, but they could be other
   appropriation (current year earnings) accounts, equity accounts, current liabilities accounts, or
   a combination thereof. The sum of these lines must balance the line(s) from the previous step.
#. Click :guilabel:`Post`.

.. example::
   Depending on the localization and the situation, the accounts involved will vary. The following
   example illustrates one potential journal entry using accounts from the Belgian :abbr:`GAAP
   (generally accepted accounting principles)` / :abbr:`CSA-WVV (code des sociétés et des
   associations - wetboek van vennootschappen en verenigingen)`. This example debits three different

>    allocation (current year earnings) accounts credits three different destination accounts (two

Reviewer: ```suggestion
   allocation (current year earnings) accounts and credits three different destination accounts (two
```

   equity accounts and one current liabilities account).

   .. list-table::
      :header-rows: 1
      :stub-columns: 1

      * - **Destination**
        - **Profit and loss appropriation account (Debit)**
        - **Balance sheet destination account (Credit)**
      * - **Legal reserve**
        - 692000 Appropriations to Legal Reserve
        - 130000 Legal Reserve
      * - **Carried Forward**
        - 693000 Profits to Be Carried Forward
        - 140000 Profit Brought Forward
      * - **Dividends**
        - 694000 Compensation for Contributions
        - 471000 Dividends - Current Financial Period

> Preparing for the Indirect Cash Flow Statement

Reviewer: ```suggestion
Preparing for the indirect cash flow statement
```
As per our [capitalization guidelines](https://www.odoo.com/documentation/master/contributing/documentation/content_guidelines.html#capitalization).

**********************************************

This new structure prevents "polluting" the generic P&L. By isolating result allocations in dedicated accounts
that the P&L report ignores for net income calculation, Odoo ensures a "pure" **Net Income** figure. This is

> vital for the **Indirect Cash Flow Statement**, which uses Net Income as its starting point to adjust for non-cash

Reviewer: ```suggestion
that the P&L report ignores for net income calculation, Odoo ensures a "pure" **net income** figure. This is
vital for the **indirect cash flow statement**, which uses net income as its starting point to adjust for non-cash
```

movements.

.. important::

>    **Action Required**: For consistency in your reporting history, it is recommended to revisit past fiscal years

Reviewer: ```suggestion
  For consistency in your reporting history, **it is recommended** to revisit past fiscal years
```

.. _accounting/year-end/current-year-earnings:

> Odoo 19.0: The New Result Allocation Logic

Reviewer 2: Since it's part of the Odoo19.0 documentation, we can probably remove "Odoo 19.0: ", which is not bringing any additional information.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

From Implicit to Explicit: The "Current Year Earnings" Type
***********************************************************

Historically, Odoo users associated the end-of-year result with account 999999. However,
the system logic is actually driven by the account type *Current Year Earnings*.

- **Previous Versions**: The result was an automatic calculation, often mixing system-generated data
  with manual entries of the same account.
- **v19.0 Paradigm**: Odoo now displays the result of the year **N** as a **Dynamic Line** called

>   "Undistributed Profits/Losses" in the Trial Balance and General Ledger of the year **N+1**. To clear

Reviewer 2: within the Trial Balance and General ledger

- **P&L accounts** reset to zero at the start of the year.

The **Dynamic Line** compensates for this timing difference. By posting an explicit entry to a
Current Year Earnings account at year-end, you replace the virtual calculation (from previous versions)
with a physical entry, "neutralizing" the dynamic line and ensuring the Balance Sheet is balanced by
actual journal items.

Affectation and Withdrawal Accounts (P&L Impact)
************************************************

In specific legislations like Belgium, the result must be allocated (to reserves, dividends, etc.)
within the same fiscal year, on "Allocation" or "Withdrawal" accounts.

- **No Carry Forward**: Like P&L accounts, allocation and withdrawal accounts do not have opening balances.
- **P&L Presentation**: These accounts **do not appear on the Balance Sheet**. When they are displayed in
  reports, they appear at the bottom of the **Profit & Loss** statement.
- **Balance Sheet Counterparts**: Only the *counterparty* of these movements (e.g., the Dividends Payable or
  the Reserve accounts) appears on the Balance Sheet. This maintains a strict separation between the process of

>   distributing profit and the resulting liabilities or equity.

Author: Yeah it is just miscellaneous points of information about these accounts. And I think that there are specific accounts from the CoA but I am not sure so I asked :)
